Content.Server/GameObjects/Components/Stack/StackComponent.cs

```diff
@@ -23,9 +24,34 @@ public class StackComponent : SharedStackComponent, IInteractUsing, IExamine
         public override int Count
         {
             get => base.Count;
-            set => base.Count = value;
+            set
+            {
+                base.Count = value;
+                
+                if (base.Count != 0 && SpriteStates != null && Owner.TryGetComponent<SpriteComponent>(out var spriteComponent))
+                {
+                    if(Count == MaxCount)
+                    {
+                        spriteComponent.LayerSetState(0, SpriteStates[2]);
+                    }
+                    else if (Count == 1)
+                    {
+                        spriteComponent.LayerSetState(0, SpriteStates[0]);
+                    }
+                    else
+                    {
+                        spriteComponent.LayerSetState(0, SpriteStates[1]);
+                    }
+                }
+            }
         }
 
+        public override string[] SpriteStates
```

Comment:
why are you overriding `SpriteStates`? You should be able to use it eitherway since you inherit the sharedcomponent